mobile/src/main/java/org/openhab/habdroid/ui/OpenHABWidgetAdapter.java

```diff
@@ -664,47 +668,82 @@ public void onItemClick(AdapterView<?> parent, View view, int index, long id) {
-                        float currentValue = setPointWidget.getItem().getStateAsFloat();
-                        currentValue = currentValue - setPointWidget.getStep();
-                        if (currentValue < setPointWidget.getMinValue())
-                            currentValue = setPointWidget.getMinValue();
-                        if (currentValue > setPointWidget.getMaxValue())
-                            currentValue = setPointWidget.getMaxValue();
-                        sendItemCommand(setPointWidget.getItem(), String.valueOf(currentValue));
+                    final Context context = getContext();
 
-                    }
-                });
-                setPointPlusButton.setOnClickListener(new OnClickListener() {
-                    public void onClick(View v) {
-                        Log.d(TAG, "Plus");
-                        OpenHABWidget setPointWidget = (OpenHABWidget) v.getTag();
-                        float currentValue = setPointWidget.getItem().getStateAsFloat();
-                        currentValue = currentValue + setPointWidget.getStep();
-                        if (currentValue < setPointWidget.getMinValue())
-                            currentValue = setPointWidget.getMinValue();
-                        if (currentValue > setPointWidget.getMaxValue())
-                            currentValue = setPointWidget.getMaxValue();
-                        sendItemCommand(setPointWidget.getItem(), String.valueOf(currentValue));
-                    }
-                });
+
+                    widgetView.setOnClickListener( new OnClickListener(){
+                        @Override
+                        public void onClick(final View view) {
+
+                            int minValue = (int) openHABWidget.getMinValue();
+                            int maxValue = (int) openHABWidget.getMaxValue();
+                            final int stepSize;
+                            if(minValue == maxValue) {
+                                stepSize = 1;
+                            } else {
+                                stepSize = (int) openHABWidget.getStep();
+                            }
+
+                            final String[] stepValues = new String[(maxValue - minValue)/stepSize +1];
+                            for(int i = 0; i < stepValues.length; i++){
+                                stepValues[i] = String.valueOf(minValue + (i*stepSize));
+                            }
+
+                            AlertDialog.Builder builder = new AlertDialog.Builder(context);
+
+                            if (labelTextView != null) {
+                                builder.setTitle(labelTextView.getText());
+                            }
+                            final LayoutInflater inflater = LayoutInflater.from(context);
+                            final View dialogView = inflater.inflate(R.layout.openhab_dialog_numberpicker, null);
```

Comment:
Android Studio suggest not to use null here

Reply:
Thanks!

If you read the link android studio send you to it says this at the end:

>  There are of course instances where you can truly justify a null parent during inflation, but they are few. One such instance occurs when you are inflating a custom layout to be attached to an AlertDialog.

That is what we are doing here. 

> The issue here is that AlertDialog.Builder supports a custom view, but does not provide an implementation of setView() that takes a layout resource; so you must inflate the XML manually. However, because the result will go into the dialog, which does not expose its root view (in fact, it doesn’t exist yet), we do not have access to the eventual parent of the layout, so we cannot use it for inflation. It turns out, this is irrelevant, because AlertDialog will erase any LayoutParams on the layout anyway and replace them with match_parent.

So I think using null there is the appropriate course of action.